pkg/storegateway/bucket_index_postings.go

```diff
@@ -87,28 +153,28 @@ func toPostingGroup(lvr indexheader.Reader, m *labels.Matcher) (*postingGroup, e
 	// have the label name set too. See: https://github.com/prometheus/prometheus/issues/3575
 	// and https://github.com/prometheus/prometheus/pull/3578#issuecomment-351653555.
 	if m.Matches("") {
-		vals, err := lvr.LabelValues(m.Name, not(m.Matches))
-		toRemove := make([]labels.Label, len(vals))
-		for i := range vals {
-			toRemove[i] = labels.Label{Name: m.Name, Value: vals[i]}
-		}
-		return newPostingGroup(true, nil, toRemove), err
+		return newLazyPostingGroup(true, m.Name, not(m.Matches))
```

Comment:
[nit] Have you considered `newLazySubtractingPostingGroup()` and `newLazyIntersectingPostingGroup()` to keep consistency with the non-lazy version and more clarity?